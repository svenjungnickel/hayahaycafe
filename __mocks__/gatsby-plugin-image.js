const React = require('react');
const plugin = jest.requireActual('gatsby-plugin-image');

const mockImage = ({ imgClassName, ...props }) =>
    React.createElement('img', {
        ...props,
        className: imgClassName,
    });

module.exports = {
    ...plugin,
    getImage: jest.fn().mockImplementation(() => false),
    getSrc: jest.fn().mockImplementation(() => false),
    GatsbyImage: jest.fn().mockImplementation(mockImage),
    StaticImage: jest.fn().mockImplementation(mockImage),
};
