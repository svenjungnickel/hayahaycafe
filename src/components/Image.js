import React from 'react';
import PropTypes from 'prop-types';
import Img from 'gatsby-image';

const Image = props => {
    const { src } = props;

    if (!!src && !!src.childImageSharp) {
        if (src.childImageSharp.fluid) {
            return <Img fluid={src.childImageSharp.fluid} {...props} />;
        }

        return <Img fixed={src.childImageSharp.fixed} {...props} />;
    }

    return <img src={src} loading="lazy" {...props} />;
};

Image.propTypes = {
    src: PropTypes.oneOfType([PropTypes.object, PropTypes.string]).isRequired,
};

export default Image;
