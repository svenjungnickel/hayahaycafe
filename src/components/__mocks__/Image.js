import React from 'react';

const mockImage = () => <img src="src" alt="alt" />;

export default mockImage;
