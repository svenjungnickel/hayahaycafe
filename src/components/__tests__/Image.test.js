import React from 'react';
import renderer from 'react-test-renderer';
import { getImage } from 'gatsby-plugin-image';
import Image from '../Image';

describe('Image', () => {
    it('renders <img>', () => {
        const component = <Image src={'src'} alt="alt" />;
        const tree = renderer.create(component).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('renders GatsbyImage', () => {
        getImage.mockImplementation(() => true);

        const component = <Image src={'image'} alt="alt" />;
        const tree = renderer.create(component).toJSON();

        expect(tree).toMatchSnapshot();
    });
});
