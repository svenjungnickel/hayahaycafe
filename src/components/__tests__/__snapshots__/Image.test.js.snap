// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image renders <img> 1`] = `
<img
  loading="lazy"
  src="src"
/>
`;

exports[`Image renders childImageSharp fixed 1`] = `
<div
  className=" gatsby-image-wrapper"
  style={
    Object {
      "display": "inline-block",
      "height": 250,
      "overflow": "hidden",
      "position": "relative",
      "width": 250,
    }
  }
>
  <picture>
    <source
      srcSet="srcSet"
    />
    <img
      alt=""
      height={250}
      loading="lazy"
      onLoad={[Function]}
      src="src"
      srcSet="srcSet"
      style={
        Object {
          "height": "100%",
          "left": 0,
          "objectFit": "cover",
          "objectPosition": "center",
          "opacity": 0,
          "position": "absolute",
          "top": 0,
          "transition": "opacity 500ms",
          "width": "100%",
        }
      }
      width={250}
    />
  </picture>
  <noscript
    dangerouslySetInnerHTML={
      Object {
        "__html": "<picture><source srcset=\\"srcSet\\" /><img loading=\\"lazy\\" width=\\"250\\" height=\\"250\\" srcset=\\"srcSet\\" src=\\"src\\" alt=\\"\\" style=\\"position:absolute;top:0;left:0;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center\\"/></picture>",
      }
    }
  />
</div>
`;

exports[`Image renders childImageSharp fluid 1`] = `
<div
  className=" gatsby-image-wrapper"
  style={
    Object {
      "maxHeight": null,
      "maxWidth": null,
      "overflow": "hidden",
      "position": "relative",
    }
  }
>
  <div
    aria-hidden={true}
    style={
      Object {
        "paddingBottom": "100%",
        "width": "100%",
      }
    }
  />
  <picture>
    <source
      sizes="(max-width: 250px) 100vw, 250px"
      srcSet="srcSet"
    />
    <img
      alt=""
      loading="lazy"
      onLoad={[Function]}
      sizes="(max-width: 250px) 100vw, 250px"
      src="src"
      srcSet="srcSet"
      style={
        Object {
          "height": "100%",
          "left": 0,
          "objectFit": "cover",
          "objectPosition": "center",
          "opacity": 0,
          "position": "absolute",
          "top": 0,
          "transition": "opacity 500ms",
          "width": "100%",
        }
      }
    />
  </picture>
  <noscript
    dangerouslySetInnerHTML={
      Object {
        "__html": "<picture><source srcset=\\"srcSet\\" sizes=\\"(max-width: 250px) 100vw, 250px\\" /><img loading=\\"lazy\\" sizes=\\"(max-width: 250px) 100vw, 250px\\" srcset=\\"srcSet\\" src=\\"src\\" alt=\\"\\" style=\\"position:absolute;top:0;left:0;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center\\"/></picture>",
      }
    }
  />
</div>
`;
